api/v1/aerospikecluster_validating_webhook.go

```diff
@@ -1281,82 +1251,41 @@ func getNamespaceReplicationFactor(nsConf map[string]interface{}) (int, error) {
 	return rf, nil
 }
 
-func validateSecurityConfigUpdate(
-	newVersion, oldVersion string, newSpec, oldSpec, currentStatus *AerospikeConfigSpec) error {
+func validateSecurityConfigUpdate(newSpec, oldSpec, currentStatus *AerospikeConfigSpec) error {
 	if currentStatus != nil {
-		currentSecurityConfig, err := IsSecurityEnabled(oldVersion, currentStatus)
+		currentSecurityEnabled, err := IsSecurityEnabled(currentStatus)
 		if err != nil {
 			return err
 		}
 
-		desiredSecurityConfig, err := IsSecurityEnabled(newVersion, newSpec)
+		desiredSecurityEnabled, err := IsSecurityEnabled(newSpec)
 		if err != nil {
 			return err
 		}
 
-		if currentSecurityConfig && !desiredSecurityConfig {
+		if currentSecurityEnabled && !desiredSecurityEnabled {
 			return fmt.Errorf("cannot disable cluster security in running cluster")
 		}
 	}
 
-	nv, err := lib.CompareVersions(newVersion, "5.7.0")
-	if err != nil {
-		return err
-	}
-
-	ov, err := lib.CompareVersions(oldVersion, "5.7.0")
-	if err != nil {
-		return err
-	}
-
-	if nv >= 0 || ov >= 0 {
-		return validateSecurityContext(newVersion, oldVersion, newSpec, oldSpec)
-	}
-
-	return validateEnableSecurityConfig(newSpec, oldSpec)
-}
-
-func validateEnableSecurityConfig(newConfSpec, oldConfSpec *AerospikeConfigSpec) error {
-	newConf := newConfSpec.Value
-	oldConf := oldConfSpec.Value
-
-	oldSec, oldSecConfFound := oldConf["security"]
-	if !oldSecConfFound {
-		return nil
-	}
-
-	newSec, newSecConfFound := newConf["security"]
-	if !newSecConfFound {
-		return fmt.Errorf("cannot remove cluster security config")
-	}
-
-	oldSecFlag, oldEnableSecurityFlagFound := oldSec.(map[string]interface{})["enable-security"]
-	newSecFlag, newEnableSecurityFlagFound := newSec.(map[string]interface{})["enable-security"]
-
-	if oldEnableSecurityFlagFound && oldSecFlag.(bool) && (!newEnableSecurityFlagFound || !newSecFlag.(bool)) {
-		return fmt.Errorf("cannot disable cluster security in running cluster")
-	}
-
-	return nil
+	return validateSecurityContext(newSpec, oldSpec)
 }
 
-func validateSecurityContext(
-	newVersion, oldVersion string, newSpec, oldSpec *AerospikeConfigSpec) error {
-	ovflag, err := IsSecurityEnabled(oldVersion, oldSpec)
+func validateSecurityContext(newSpec, oldSpec *AerospikeConfigSpec) error {
+	ovflag, err := IsSecurityEnabled(oldSpec)
 	if err != nil {
 		if !errors.Is(err, internalerrors.ErrNotFound) {
```

Comment:
`IsSecurityEnabled` will never return `ErrNotFound` type error, this is being handled in the `IsSecurityEnabled` defination